ts/src/provider.ts

```diff
@@ -195,39 +228,228 @@ export default class Provider {
    * Simulates the given transaction, returning emitted logs from execution.
    *
    * @param tx      The transaction to send.
-   * @param signers The set of signers in addition to the provdier wallet that
-   *                will sign the transaction.
+   * @param signers The signers of the transaction.
    * @param opts    Transaction confirmation options.
    */
   async simulate(
     tx: Transaction,
-    signers?: Array<Signer | undefined>,
-    opts: ConfirmOptions = this.opts
-  ): Promise<RpcResponseAndContext<SimulatedTransactionResponse>> {
-    if (signers === undefined) {
-      signers = [];
-    }
-
+    signers?: Signer[],
+    commitment?: Commitment,
+    includeAccounts?: boolean | PublicKey[]
```

Comment:
Do we need this `includeAccounts` var? What is it used for? Seems like we can remove it.

Reply:
It lets the caller indicate whether they are interested in the account states after the simulation. I assume this option exists to reduce RPC load in cases where it is not needed